client/src/partials/reports_proposed/variation_exploitation/variation_exploitation.js

```diff
@@ -0,0 +1,69 @@
+var variationExploitationController = function ($http, $translate, validate, reportConfigService, messenger) {
+  var dependencies = {}, generatedDocumentPath = null, serverUtilityPath = 'report/build/variation_exploitation', 
+    configuration = reportConfigService.configuration, vm = this;
+
+  dependencies.fiscalYears = {
+    query : {
+      identifier : 'id',
+      tables : {
+        'fiscal_year' : {
+          columns : ['id', 'fiscal_year_txt', 'previous_fiscal_year']
+        }
+      }
+    }
+  };
+
+  vm.generate_doc = $translate.instant('VARIATION_EXPLOITATION.GENERATE_DOC');
+  vm.loading = $translate.instant('VARIATION_EXPLOITATION.LOADING');
+  vm.building = false;
+  vm.configuration = configuration;
+
+  validate.process(dependencies)
+    .then(setDefaultConfiguration);
+
+  function setDefaultConfiguration(models) {
+    angular.extend(vm, models);
+    selectConfiguration('language', configuration.language.options[1]);
+    vm.fiscal_year_id = vm.fiscalYears.data[vm.fiscalYears.data.length-1].id;
+    vm.previous_fiscal_year_id = vm.fiscalYears.data[vm.fiscalYears.data.length-1].previous_fiscal_year;
+  }
+
+  function selectConfiguration(key, value) {
+    configuration[key].selected = value;
+  }
+
+  function generateDocument() {
+    var path = serverUtilityPath;
+    var configurationObject = {};
+
+    configurationObject.language = configuration.language.selected.value;
+    configurationObject.fy = vm.fiscal_year_id;
+    configurationObject.pfy = vm.previous_fiscal_year_id ||  vm.fiscal_year_id;
+    configurationObject.enterprise = configuration.enterprise;
+    configurationObject.project = configuration.project;
+
+    vm.building = true;
+
+    $http.post(path, configurationObject)
+      .success(function (result) {
+        // Expose generated document path to template
+        vm.building = false;
+        vm.generatedDocumentPath = result;
+      })
+      .error(function (code) {
+        vm.building = false;
```

Comment:
You've written `vm.building = false;` twice, in the `.success()` and `.error()` methods.  If `vm.building` is independent of the result of the promise, you should put it in a `.finally()` method, which will by called last to clean the promise up.